File: solutions/cpp/princesspeach.cpp
// https://open.kattis.com/problems/princesspeach
// Difficulty: 2.2 (Easy)

#include <iostream>
#include <vector>
using namespace std;

int main()
{
    int totalNumbers;
    cin >> totalNumbers;

    int totalFound;
    cin >> totalFound;

    vector<int> found;

    int mistakes = 0;
    for(int i = 0; i < totalFound; i++)
    {
        bool foundMistake = false;
        int foundNum;
        cin >> foundNum;

        for(int j = 0; j < found.size(); j++)
        {
            if (foundNum == found.at(j))
            {
                foundMistake = true;
                mistakes++;
                break;
            }
            else
            {
                continue;
            }
        }

        if (!foundMistake)
        {
            found.push_back(foundNum);
        }
    }

    for(int i = 0; i < totalNumbers; i++)
    {
        bool foundMistake = false;
        for(int j = 0; j < found.size(); j++)
        {
            if (i == found[j])
            {
                foundMistake = true;
                break;
            }
            else
            {
                continue;
            }
        }

        if (!foundMistake)
        {
            cout << i << endl;
        }
    }

    cout << "Mario got " << totalFound - mistakes << " of the dangerous obstacles." <<endl;

    return 0;
}

File: solutions/java/princesspeach.java
// https://open.kattis.com/problems/princesspeach
// Difficulty: 2.2 (Easy)

import java.util.Scanner;
import java.util.ArrayList;

public class princesspeach
{
    public static void main(String[] args)
    {
        Scanner scn = new Scanner(System.in);
        int totalNumbers = scn.nextInt();
        int totalFound = scn.nextInt();

        ArrayList<Integer> found = new ArrayList<Integer>();
        int mistakes = 0;
        for(int i = 0; i < totalFound; i++)
        {
            int num = scn.nextInt();
            if (found.contains(num))
            {
                mistakes++;
            }
            else
            {
                found.add(num);
            }
        }

        for(int i = 0; i < totalNumbers; i++)
        {
            if (!found.contains(i))
            {
                System.out.println(i);
            }
        }

        System.out.printf("Mario got %s of the dangerous obstacles.\n", totalFound - mistakes);

        scn.close();
    }
}